deeprankcore/dataset.py

```diff
@@ -685,7 +693,30 @@ def __init__( # pylint: disable=too-many-arguments, too-many-locals
                 dataset_train._compute_mean_std()
             self.means = dataset_train.means
             self.devs = dataset_train.devs
-
+            
+    def _check_inherited_param(
+        self,
+        inherited_param: List[str] = None,
+        dataset_train: GraphDataset = None,
+    ):
+        """Check if the parameters from validation and testing are same as the training set.
+        
+        Args:
+        inherited_param (List[str], optional): List of parameters that need to be check for inheritance.
+            Defaults to None.
+        dataset_train (class:`GraphDataset`, optional): The parameters in `inherited_param` will be inherited from `dataset_train`.
+            Defaults to None.
+        """
+        
+        self_vars = vars(self)
```

Comment:
This could be further simplified (line 711 - 718):

```python
        for param in inherited_param:
            train_param = getattr(dataset_train, param)
            if getattr(self, param) != train_param:
                setattr(self, param, train_param)
                _log.warning(f"Overwriting {param} parameter with the one " +
                            f"used in the training phase: {train_param}.")   
```